src/mlpack/core/optimizers/sgd/update_policies/vanilla_update.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file vanilla_update.hpp
+ * @author Arun Reddy
+ *
+ * Empty update for SGD
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SGD_EMPTY_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGD_EMPTY_UPDATE_HPP
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Vanilla update policy for SGD.
+ *
+ */
+class VanillaUpdate {
+ public:
+
+  /**
+   * Function for runtime initialization.
+   *
+   * @param n_rows
+   * @param n_cols
+   */
+  void Initialize(const size_t n_rows,
+                  const size_t n_cols)
+  {/* Do Nothing */}
+
+  arma::mat Update(double stepSize, arma::mat gradient)
+  {
+    // perform the vanilla SGD update.
+    return - stepSize * gradient;
```

Comment:
Oh, yeah, I missed the pass-by-value of the gradient, that should definitely be `const arma::mat&`.  I think for `MomentumPolicy` the compiler would be smart enough to avoid the unnecessary copy, but I don't think that will apply here, so I agree, `void Update(arma::mat& iterate, const double stepSize, const arma::mat& gradient)` would be the right way to go.